### PR TITLE
feat: Implement eth.calculateFeeData 

### LINKED ISSUE #6599: Implement `provider.getFeeData()`
I suggest to implement `provider.getFeeData()` similar to https://docs.alchemy.com/reference/sdk-getfeedata and https://docs.ethers.org/v6/single-page/#Provider-getFeeData.

From Alchemy API Reference:
> Returns the recommended fee data to use in a transaction. For an EIP-1559 transaction, the maxFeePerGas and maxPriorityFeePerGas should be used.
> 
> For legacy transactions and networks which do not support EIP-1559, the gasPrice should be used.

And deprecate getGasPrice  (check https://github.com/ethers-io/ethers.js/discussions/2439#discussioncomment-1857403)

Helpful references:
- https://www.quicknode.com/guides/ethereum-development/transactions/how-to-send-an-eip-1559-transaction
- https://ethereum-magicians.org/t/eip-1559-fee-market-change-for-eth-1-0-chain/2783/310

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #6599

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by Muhammad-Altabba** [comment]
It seems there is a cashing problem that lead to report old errors that has been already resolved. I overcome this by requesting all the tests to rerun.

**Review by avkos — APPROVED** [review]

File: packages/web3-eth/src/web3_eth.ts [hunk 1]
```diff
@@ -238,6 +242,99 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
...
+	 *
+	 * ```ts
+	 * web3.eth.getMaxPriorityFeePerGas().then(console.log);
+	 * > 20000000000n
+	 *
+	 * web3.eth.getMaxPriorityFeePerGas({ number: FMT_NUMBER.HEX , bytes: FMT_BYTES.HEX }).then(console.log);
+	 * > "0x4a817c800"
+	 * ```
+	 */
+	public async getMaxPriorityFeePerGas<
```
Comment (jdevcs) [inline packages/web3-eth/src/web3_eth.ts]: will be great to add e2e tests for this as well for geth.

File: packages/web3-eth/src/web3_eth.ts [hunk 2]
```diff
@@ -238,6 +242,99 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
...
+	 * web3.eth.getFeeData(ethUnitMap.Gwei, 2n).then(console.log);
+	 * > {
+	 *     gasPrice: 20000000000n,
+	 *     maxFeePerGas: 40000000000n,
+	 *     maxPriorityFeePerGas: 20000000000n,
+	 * 	   baseFeePerGas: 20000000000n
+	 * }
+	 * ```
+	 */
+	public async calculateFeeData(
```
Comment (jdevcs) [inline packages/web3-eth/src/web3_eth.ts:290]: and for this e2e test

File: packages/web3-types/src/eth_types.ts [hunk 3]
```diff
@@ -529,3 +529,48 @@ export interface Eip712TypedData {
...
+ * EIP-1559 was applied to Ethereum after London hardfork.
+ *  
+ * Typically you will only need `maxFeePerGas` and `maxPriorityFeePerGas` for a transaction following EIP-1559.
+ * However, if you want to get informed about the fees of last block, you can use `baseFeePerGas` too.
+ * 
+ * 
+ * 	@see https://eips.ethereum.org/EIPS/eip-1559
+ * 
+ */
+export interface FeeData {
```
Comment (jdevcs) [inline packages/web3-types/src/eth_types.ts:544]: changelog of web3-types needs update

**Comment by jdevcs** [comment]
> # Bundle Stats
> Hey there, this message comes from a github action that helps you and reviewers to understand how these changes affect the size of this project's bundle.
> 
> As this PR is updated, I'll keep you updated on how the bundle size is impacted.
> 
> **Total**
> 
> Asset	Old size	New size	Diff	Diff %
> Total	0	619 KB	619 KB	-
> View detailed bundle breakdown

why its not working

**Comment by jdevcs** [comment]
codecov report & actions are also not triggered for this PR,

**Review by jdevcs — COMMENTED** [review]
, LGTM, just need some tests / added some questions, Thanks Muhammad.

**Comment by Muhammad-Altabba** [comment]
> codecov report & actions are also not triggered for this PR,

I do not know why for sure. But it seems because I re-triggered all CI actions (https://github.com/web3/web3.js/pull/6795#issuecomment-1932046069)

File: packages/web3-eth/src/web3_eth.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-eth/src/web3_eth.ts]: Will be added in: https://github.com/web3/web3.js/pull/6748

File: packages/web3-types/src/eth_types.ts (hunk 3 above)
Comment (Muhammad-Altabba) [reply packages/web3-types/src/eth_types.ts:544]: done

**Review by luu-alex — APPROVED** [review]

**Pull request merged**